src/bp/ui-shared/src/TreeView/index.tsx

```diff
@@ -0,0 +1,288 @@
+import { Classes, ContextMenu, ITreeNode, Tree } from '@blueprintjs/core'
+import _ from 'lodash'
+import React, { useEffect, useReducer, useState } from 'react'
+
+export const FOLDER_ICON = 'folder-close'
+export const DOCUMENT_ICON = 'document'
+
+export interface TreeViewProps<T> {
```

Comment:
Can you put the interfaces in a separate file inside the TreeView folder? It would declutter this file